### PR TITLE
CI: report failures on discord

### PR DESCRIPTION
**Pull request by yannbf**
Issue:

## What I did

Set up a reporting mechanism that will post updates to discord on workflow failures:

image

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -175,6 +176,30 @@ commands:
...
+        description: |
+          Which template to report in discord
+        type: string
+    steps:
+      - run:
+          when: on_fail
+          command: git fetch --unshallow
+      - list-recent-merges
+      - discord/status:
+          only_for_branches: chore/failure-reports
```
Comment (kasperpeulen) [inline .circleci/config.yml:200]: This is nice, I think if we do this only for the `next` and `alpha` branch, we have exactly what we want.

I first thought about only notifying on the `merged` and `daily` workflow, but that would also notify when you run the `daily` workflow in your PR.

So setting this to `next would be better I guess.

**Review by kasperpeulen — APPROVED** [review]
This is awesome!

File: .circleci/config.yml (hunk 1 above)
Comment (yannbf) [reply .circleci/config.yml]: I had a similar thought and I think it makes sense too, I'll leave both next and alpha branches there, possibly updating to main in the future too 🤔

**Review by JReinhold — APPROVED** [review]
Looks great!

Two suggestions.

1. Would it make sense to include in the message if this was `merged` or `daily` that failed? Would make it easier to narrow down when the issue was introduced
2. similarly, it might be easier to debug errors in `merged` if they only included the PR that actually got merged, and not the most recent ones.

EXCEPT if new merges cancels eachother out, I don't know. Eg. if Yann merges something to next, and I do too 3 minutes later. If my merge cancels Yann's, then an error in Yann's merge will be caught in mine, and so we would need all the recent PRs. If that is the case, disregard what I wrote.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
> new merges cancels each other out
yes they do.

**Comment by kasperpeulen** [comment]
@ndelangen @JReinhold 
They are canceled, with the exception for the default branch (next):
image

**Pull request merged**